dto/src/main/java/ca/on/oicr/gsi/runscanner/dto/IlluminaNotificationDto.java

```diff
@@ -22,6 +25,17 @@ public class IlluminaNotificationDto extends NotificationDto {
   private String workflowType;
   private IndexSequencing indexSequencing;
 
+  public Map<Integer, Analysis> analysis = new HashMap<>();
```

Comment:
I'd expect this to serialize something like this:

```
"analysis": {
  1: { ... (Analysis object) },
  2: { ... (Analysis object) }
}
```

That's not valid JSON (integer keys), so I'm guessing that's the serialization issue, though I'm surprised it doesn't throw an exception.

A possible good solution is to store the attempt number as a field in `Analysis` and make `IlluminaNotificationDto.analysis` a `List<Analysis>` or similar `Set`/array. This would also give a purpose to the `Analysis` interface, which right now is basically just `Object`. Could also make `Analysis` an abstract class instead of an interface.

A worse solution would be to just change the `Map` key to `String` and store the number as a string.

Reply:
Given that I am already getting the number as a String and using Integer.valueOf() to parse it at NovaseqXProcessor:50, I will just stop doing that and use Strings

Reply:
Unfortunately making this change did not fix the serialization problem

Reply:
also, even with an integer key, that part always serialized successfully, it's the value that doesn't:
![image](https://github.com/user-attachments/assets/d9993269-ba63-40fe-957f-ca4854b5cfbc)